src/axom/core/ArrayBase.hpp

```diff
@@ -552,6 +648,13 @@ class ArrayBase<T, 1, ArrayType>
     return idx >= 0 && idx < asDerived().size();
   }
   /// @}
+protected:
+  /// \brief The sizes (extents?) in each dimension
+  StackArray<IndexType, 1> m_dims;
+  /// \brief Spacing between consecutive elements
+  IndexType m_spacing;
+  /// \brief The strides in each dimension
+  StackArray<IndexType, 1> m_strides;
```

Comment:
Do we need all of these (`m_dims`, `m_spacing`, `m_strides`) for the `ArrayBase` specialization when  `DIM==1`? Perhaps we can use member functions to return some of these values?

Reply:
My concern is in the memory overhead associated with an `axom::Array` and `axom::ArrayView` compared to a `std::vector` (and eventually a `std::range`). Especially since we pass `axom::ArrayView`s by value.

Reply:
@kennyweiss I understand.  Maybe `m_spacing` can be removed (as @publixsubfan suggested) if we replace the `inBounds` method with `inLogicalBounds` and `inMemoryBounds` and use the appropriate one.  I'll look more closely at the 1D case to try to remove the `m_dims` and `m_strides` from it.

Reply:
@kennyweiss I added the documentations.  I also found a way to restore the stateless 1D `ArrayBase` implementation.  I moved the spacing parameter into the child classes, and had the base class access it through `asDerived()`.  This had the added benefit of letting `Array` hardwire the spacing to 1, so the compilers could optimize it away.  Only `ArrayView` need to have spacing determined at run time.